### PR TITLE
Fix get_type_description service bug and add a unit test

### PR DESCRIPTION
**Pull request by emersonknapp**
It actually just crashed when called, not sure how I let that through.
This fixes the issue and adds a unit test to try out the service.

Will need backport to Iron.

### CODE REVIEW

File: rclpy/rclpy/type_description_service.py [hunk 1]
```diff
@@ -40,6 +40,7 @@ class TypeDescriptionService:
     def __init__(self, node):
         """Initialize the service, if the parameter is set to true."""
         node_name = node.get_name()
+        self._node = node
```
Comment (clalancette) [inline rclpy/rclpy/type_description_service.py:43]: So if we are setting `self._node` here so we can use it later in `_service_callback`, then presumably we don't need the `node` parameter to `_start_service`, right?  We can just use `self._node` there as well, I think?

Comment (emersonknapp) [reply rclpy/rclpy/type_description_service.py:43]: That's right

**Comment by emersonknapp** [comment]
Pulls: ros2/rclpy#1155
Gist: https://gist.githubusercontent.com/emersonknapp/49c02ab93657994a884d1d647949cbe9/raw/558d71014dc2b50b0179889b2d68db710f058f70/ros2.repos
BUILD args:  --packages-above-and-dependencies rclpy
TEST args:  --packages-above rclpy
ROS Distro: rolling
Job: ci_launcher
ci_launcher ran: https://ci.ros2.org/job/ci_launcher/12573
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/19410/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/13938/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/20132/)

**Review by clalancette — APPROVED** [review]
Looks good to me with green CI.

**Comment by emersonknapp** [comment]
Update: this was actually causing `test_rosout_subscription.py` to fail, because I was not using `weakref` of Node like the Parameter service does. Updated now and should make both `RPr` and CI green

**Comment by emersonknapp** [comment]
Pulls: ros2/rclpy#1155
Gist: https://gist.githubusercontent.com/emersonknapp/6b96a55838e2c6ae5817a135489055a9/raw/558d71014dc2b50b0179889b2d68db710f058f70/ros2.repos
BUILD args:  --packages-above-and-dependencies rclpy
TEST args:  --packages-above rclpy
ROS Distro: rolling
Job: ci_launcher
ci_launcher ran: https://ci.ros2.org/job/ci_launcher/12574
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/19411/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/13939/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/20133/)

**Comment by emersonknapp** [comment]
Note windows test failure: `rosbag2_tests.test_rosbag2_play_end_to_end.gtest.missing_result`

**Review by fujitatomoya — APPROVED** [review]

**Pull request merged**